api/src/pcapi/utils/email.py

```diff
@@ -49,3 +51,10 @@ def is_valid_email_domain(content: str) -> bool:
         return True
     except email_validator.EmailNotValidError:
         return False
+
+
+def is_email_whitelisted(email: str) -> bool:
+    # Returns True if email domain is found in WHITELISTED_EMAIL_RECIPIENTS
+    # To whitelist a domain, the format must be '*@my.domain';
+    domain = f"*@{email.split('@')[1]}" if is_valid_email(email) else None
+    return email in settings.WHITELISTED_EMAIL_RECIPIENTS or domain in settings.WHITELISTED_EMAIL_RECIPIENTS
```

Comment:
Comme tu vérifies `is_valid_email`, a priori ça ne sert à rien de continuer s'il n'est pas valide :)
Je propose :
```suggestion
    if not is_valid_email(email):
        return False
    domain = f"*@{email.split('@')[1]}"
    return email in settings.WHITELISTED_EMAIL_RECIPIENTS or domain in settings.WHITELISTED_EMAIL_RECIPIENTS
```